### PR TITLE
[LMS-82][Frontend] Course Settings

### PR DESCRIPTION
**Pull request by mikorosales**
## Issue Link
[LMS-82](https://framgiaph.backlog.com/view/LMS-82)

## Defintion of Done
ShareableLink component should be created under shared > components
RichInputTextField component should be create under shared > components
Should have similar fields base on the provided design reference

## Steps to reproduce
http://localhost:3000/trainer/course/detail/[id]

Try:
- npm install

If does not work, install these packages:

- npm install draft-js
- npm install react-draft-wysiwyg
- npm install react-icons feather-icons
- npm install react-copy-to-clipboard
-----
- npm i --save-dev @types/draft-js
- npm i --save-dev @types/react-draft-wysiwyg
- npm i --save-dev @types/react-copy-to-clipboard
-----
- If all packages are installed and still not working, you can try docker-compose stop then start again
- Refresh the page

## Affected Components / Functionalities / Page
n/a

## Test Cases
_will update soon..._

## Notes
n/a

## Screenshots
058eb41a5b231dcfce654f054e0e4d73
image

### CODE REVIEW

File: client/src/shared/components/RadioButton/index.tsx [hunk 1]
```diff
@@ -6,10 +6,18 @@ export interface Props {
...
 
-const RadioButton: FC<Props> = ({ label, options, alignment, classname, onClickEvent }: Props) => {
+const RadioButton: FC<Props> = ({
+  label,
+  options,
+  alignment,
+  classname,
+  description,
+  onClickEvent
+}: Props) => {
```
Comment (ragg09-sunasterisk) [inline client/src/shared/components/RadioButton/index.tsx:20]: please update the interface to Specific naming e.g. `RadioButtonProps`

File: client/src/shared/components/Button/index.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ const Button: React.FunctionComponent<ButtonProps> = ({
   return (
     <div className="flex flex-row">
       <button
-        className="m-2 text-white font-bold py-2 px-4 rounded border"
+        className="text-white font-bold py-2 px-4 rounded border"
```
Comment (ragg09-sunasterisk) [inline client/src/shared/components/Button/index.tsx:28]:
```suggestion
        className="mb-4 text-white font-bold py-2 px-4 rounded border"
```

**Review by ragg09-sunasterisk — COMMENTED** [review]
I cant proceed to the page
image


I followed the Steps to Reporduce
here's the current package,json for reference
image

**Comment by Marielleregner7** [comment]
Comment:
- Please add visible text box.
- Also, please check grammatical error.
mikocoursesettings

Thank you 🙇

**Comment by Marielleregner7** [comment]
Comment:
I think in the https://framgiaph.backlog.com/view/LMS-82 it is stated that the Sidebar Navigation is included. 🙇

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- [ ] Course Settings is visible and is identical to the design reference.
- [x] No ESLint warnings or errors.
- [x] No error found, when running npm run build.

Comments: 5
Test time: 135 mins
Iterations: 2

Proof:
coursesettingsmiko
image

File: client/src/shared/layouts/CourseSettingsSideBar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+const CourseSettingsSidebar: React.FunctionComponent = () => {
+  return (
+    <div className="space-y-9">
+      <div>
+        <div>
+          <h1 className="text-lg pb-2 font-medium">Basic Course Settings</h1>
+        </div>
+        <div className="space-y-1 ml-5">
+          <h3 className="underline">
+            <Link href="/">Course Name</Link>
```
Comment (ragg09-sunasterisk) [inline client/src/shared/layouts/CourseSettingsSideBar/index.tsx:13]: Please update this accordingly, clicking any anchor in sidebar must jump to its corresponding ID in course-settings

```
<a href="#exampleID">Example</a>

<div id="exampleID">
  ...
</div>
```

**Review by ragg09-sunasterisk — COMMENTED** [review]
- Dont Create a page for this
- move this to layout or section
- append this to [LMS-335][FE]Course Detail Header (https://github.com/framgia/sph-lms/pull/50)
- setting must show after clicking "Course Setting" in Tabs

File: client/src/pages/trainer/course/detail/[id].tsx [hunk 4]
```diff
@@ -87,8 +88,8 @@ const CourseDetails: React.FC = () => {
               <div className="flex justify-end pr-6">
                 <Button text={'Add Course Content'} color={'#325184'}></Button>
               </div>
-              <div className="flex justify-center items-center h-auto w-auto font font-medium text-xl">
-                Content for Tab {activeTab}
+              <div className="flex h-auto w-auto font font-medium">
+                {activeTab === 1 ? <CourseSetting /> : null}
```
Comment (ragg09-sunasterisk) [inline client/src/pages/trainer/course/detail/[id].tsx:92]: As much as possible. optimize the code by excluding empty condition. 

```suggestion
               {activeTab === 1 && <CourseSetting />}
```

**Comment by ragg09-sunasterisk** [comment]
Comments: 8
Iterations: 3

**Pull request merged**